superset/views/core.py

```diff
@@ -1513,6 +1513,25 @@ def tables(self, db_id, schema, substr, force_refresh='false'):
             table_names = [tn for tn in table_names if substr in tn]
             view_names = [vn for vn in view_names if substr in vn]
 
+        if not schema and database.default_schemas:
```

Comment:
Nice, I like this because if the parameter is not there it doesn't affect the current behavior, so people can opt-in into this feature.